### ISSUE TITLE
Error "Update mkParentDir failed" when using rclone with ocis

### ISSUE THREAD

**Issue by nikslor** [owncloud/ocis#4762]
## Describe the bug

I run into an issue when trying to sync my home dir to ocis (2.0.0) using rclone (1.61.1). On the client side I see messages like the following:

```
$ rclone -v -L sync /tmp/blackhole/ocis/ storage:/test
2023/01/31 18:04:15 ERROR : foo/bar/baz/test.txt: Failed to copy: Update mkParentDir failed: Resource not found: Sabre\DAV\Exception\NotFound: 404 Not Found
2023/01/31 18:04:15 INFO  : test.txt: Copied (new)
2023/01/31 18:04:15 ERROR : webdav root 'test': not deleting files as there were IO errors
2023/01/31 18:04:15 ERROR : webdav root 'test': not deleting directories as there were IO errors
2023/01/31 18:04:15 ERROR : Attempt 1/3 failed with 1 errors and: Update mkParentDir failed: Resource not found: Sabre\DAV\Exception\NotFound: 404 Not Found
2023/01/31 18:04:15 ERROR : foo/bar/baz/test.txt: Failed to copy: Update mkParentDir failed: Resource not found: Sabre\DAV\Exception\NotFound: 404 Not Found
```

## Steps to reproduce

* Let's assume your working dir is ```/tmp/blackhole/ocis```
* create ```test.txt```: ```echo "Hello ocis" > test.txt```
* create a simple folder structure ```mkdir -p foo/bar/baz```
* copy ```text.txt``` to that folder ```cp test.txt foo/bar/baz/```
* start sync ```rclone -v -L sync /tmp/blackhole/ocis/ storage:/test```

## Expected behavior

I should be able to sync this folder with rclone w/o seeing such an error.

## Actual behavior

I get an error with the latest OCIS stable version.

## Setup

Please describe how you started the server and provide a list of relevant environment variables or configuration files.

```did setcap 'cap_net_bind_service=+ep' ocis and run ocis like this:```

```
OCIS_BASE_DATA_PATH=/var/lib/ocis
ACCOUNTS_DEMO_USERS_AND_GROUPS=false
PROXY_HTTP_ADDR=0.0.0.0:443
OCIS_URL=https://storage.0x17.ch
PROXY_TRANSPORT_TLS_KEY=/etc/letsencrypt/live/storage.0x17.ch/privkey.pem
PROXY_TRANSPORT_TLS_CERT=/etc/letsencrypt/live/storage.0x17.ch/fullchain.pem
OCIS_INSECURE=false
PROXY_ENABLE_BASIC_AUTH=true
```

## Additional context

I had a similar issue earlier on, see: #2767

**Comment by nikslor** [comment]
@wkloucek I think this may be related to #2767, maybe something you want to have a look at?
(I hope it's fine to highlight you - otherwise just let me know)

**Comment by nikslor** [comment]
I also see this with rclone on a Windows 11 box syncing to the same server.

**Comment by micbar** [comment]
There was a fix in #6136 which is available on ocis master which might fix this issue.

**Comment by nikslor** [comment]
Hi @micbar I can confirm, that this is fixed in a recent release! I tested with 3.0.0 rc3 and the problem is gone.
Awesome! Thanks a lot for the work on this!
